stingray/lightcurve.py

```diff
@@ -70,9 +81,25 @@ def __init__(self, time, counts, input_counts=True):
         if input_counts:
             self.counts = np.asarray(counts)
             self.countrate = self.counts/self.dt
+            if err is not None:
+                assert np.all(np.isfinite(err)), "There are inf or NaN " \
+                                                 "values in your err array!"
```

Comment:
I think you should be able to pull that assertion out of the two if statements to save  you the duplication.